src/transactionComputer.ts

```diff
@@ -41,14 +43,23 @@ export class TransactionComputer {
     }
 
     computeBytesForSigning(transaction: ITransaction): Uint8Array {
-        // TODO: do some checks for the transaction e.g. sender, chain ID etc.
-        // TODO: for appropriate tx.version, interpret tx.options accordingly and sign using the content / data hash
+        this.ensureFields(transaction);
```

Comment:
We can call this `validate()` or `checkValidity()` or `ensureValid()`. And then, within it, we can also check the following constraint: `if tx.version is 1 but an option is set, raise error`.

Can be done in a separate PR.

Reply:
Will do in a future PR.